mdp/hinet/layer.py

```diff
@@ -116,7 +134,7 @@ def _get_supported_dtypes(self):
         return list(types)
 
     def is_trainable(self):
-        return any(node.is_trainable() for node in self.nodes)
+        return self._is_trainable
```

Comment:
Did you run into efficiency proplems with the old code? I'm wondering at what point this makes a measurable difference.